### ISSUE TITLE
ICE: Specialization causes panic when normalizing associated type with invalid where clause

### ISSUE THREAD

**Issue by trait-Cong** [rust-lang/rust#150857]
I tried this code:

```rust
#![feature(specialization)]
trait Assoc {
    type Output;
}
default impl<T> Assoc for T {
    type Output = bool;
}
impl Assoc for u8
where
    unDef: Assoc,
{}
trait Foo {}
impl Foo for <u8 as Assoc>::Output {}
fn main() {}

```

I expected to see this happen: The compiler should emit a standard error E0412 (cannot find type unDef in this scope) and exit gracefully.

Instead, this happened: The compiler crashed with an Internal Compiler Error (ICE).

### Meta

`rustc --version --verbose`:
```
rustc 1.94.0-nightly (e29fcf45e 2026-01-04)
binary: rustc
commit-hash: e29fcf45e4ae686d77b490bf07320f0d3a2cf35f
commit-date: 2026-01-04
host: x86_64-unknown-linux-gnu
release: 1.94.0-nightly
LLVM version: 21.1.8
```

<details><summary>Backtrace</summary>
<p>

```
thread 'rustc' panicked at /rustc-dev/e29fcf45e4ae686d77b490bf07320f0d3a2cf35f/compiler/rustc_trait_selection/src/traits/specialize/mod.rs:133:21:
Box<dyn Any>
stack backtrace:
   0:     0x7f2ad5819c0b - <std[eaa43e369792efb9]::backtrace::Backtrace>::create
   1:     0x7f2ad5819b55 - <std[eaa43e369792efb9]::backtrace::Backtrace>::force_capture
   2:     0x7f2ad48115b7 - std[eaa43e369792efb9]::panicking::update_hook::<alloc[83e7320d2ee5a6b1]::boxed::Box<rustc_driver_impl[5af842e777d449e6]::install_ice_hook::{closure#1}>>::{closure#0}
   3:     0x7f2ad582c4b2 - std[eaa43e369792efb9]::panicking::panic_with_hook
   4:     0x7f2ad484de81 - std[eaa43e369792efb9]::panicking::begin_panic::<rustc_errors[3cfa44cd230a06eb]::ExplicitBug>::{closure#0}
   5:     0x7f2ad483e4e6 - std[eaa43e369792efb9]::sys::backtrace::__rust_end_short_backtrace::<std[eaa43e369792efb9]::panicking::begin_panic<rustc_errors[3cfa44cd230a06eb]::ExplicitBug>::{closure#0}, !>
   6:     0x7f2ad483e11c - std[eaa43e369792efb9]::panicking::begin_panic::<rustc_errors[3cfa44cd230a06eb]::ExplicitBug>
   7:     0x7f2ad486bd21 - <rustc_errors[3cfa44cd230a06eb]::diagnostic::BugAbort as rustc_errors[3cfa44cd230a06eb]::diagnostic::EmissionGuarantee>::emit_producing_guarantee
   8:     0x7f2ad4e19ea9 - rustc_middle[829e5443847148d]::util::bug::opt_span_bug_fmt::<rustc_span[14443e5baae9ab13]::span_encoding::Span>::{closure#0}
   9:     0x7f2ad4e1a032 - rustc_middle[829e5443847148d]::ty::context::tls::with_opt::<rustc_middle[829e5443847148d]::util::bug::opt_span_bug_fmt<rustc_span[14443e5baae9ab13]::span_encoding::Span>::{closure#0}, !>::{closure#0}
  10:     0x7f2ad4e0b65b - rustc_middle[829e5443847148d]::ty::context::tls::with_context_opt::<rustc_middle[829e5443847148d]::ty::context::tls::with_opt<rustc_middle[829e5443847148d]::util::bug::opt_span_bug_fmt<rustc_span[14443e5baae9ab13]::span_encoding::Span>::{closure#0}, !>::{closure#0}, !>
  11:     0x7f2ad2863aa4 - rustc_middle[829e5443847148d]::util::bug::bug_fmt
  12:     0x7f2ad6097fcd - rustc_trait_selection[feb4a4f4d9146cdf]::traits::project::confirm_candidate
  13:     0x7f2ad6077c8b - rustc_trait_selection[feb4a4f4d9146cdf]::traits::project::normalize_projection_term
  14:     0x7f2ad60758de - <rustc_trait_selection[feb4a4f4d9146cdf]::traits::normalize::AssocTypeNormalizer as rustc_type_ir[241df8ba430ad1e5]::fold::TypeFolder<rustc_middle[829e5443847148d]::ty::context::TyCtxt>>::fold_ty
  15:     0x7f2ad6075354 - <&rustc_middle[829e5443847148d]::ty::list::RawList<(), rustc_middle[829e5443847148d]::ty::generic_args::GenericArg> as rustc_type_ir[241df8ba430ad1e5]::fold::TypeFoldable<rustc_middle[829e5443847148d]::ty::context::TyCtxt>>::fold_with::<rustc_trait_selection[feb4a4f4d9146cdf]::traits::normalize::AssocTypeNormalizer>
  16:     0x7f2ad5eb230f - <rustc_trait_selection[feb4a4f4d9146cdf]::traits::select::SelectionContext>::match_impl::{closure#0}
  17:     0x7f2ad6da402d - <rustc_trait_selection[feb4a4f4d9146cdf]::traits::select::SelectionContext>::assemble_candidates_from_impls
  18:     0x7f2ad5e8eeb9 - <rustc_trait_selection[feb4a4f4d9146cdf]::traits::select::SelectionContext>::candidate_from_obligation::{closure#0}::{closure#0}
  19:     0x7f2ad70c85bb - <rustc_trait_selection[feb4a4f4d9146cdf]::traits::select::SelectionContext>::poly_select::{closure#0}
  20:     0x7f2ad658017a - <rustc_trait_selection[feb4a4f4d9146cdf]::traits::fulfill::FulfillProcessor as rustc_data_structures[4e9b1f6e439fd3ef]::obligation_forest::ObligationProcessor>::process_obligation
  21:     0x7f2ad5e046e4 - <rustc_data_structures[4e9b1f6e439fd3ef]::obligation_forest::ObligationForest<rustc_trait_selection[feb4a4f4d9146cdf]::traits::fulfill::PendingPredicateObligation>>::process_obligations::<rustc_trait_selection[feb4a4f4d9146cdf]::traits::fulfill::FulfillProcessor>
  22:     0x7f2ad669cd72 - <rustc_trait_selection[feb4a4f4d9146cdf]::traits::fulfill::FulfillmentContext<rustc_trait_selection[feb4a4f4d9146cdf]::traits::FulfillmentError> as rustc_infer[de7e637fb9d30ade]::traits::engine::TraitEngine<rustc_trait_selection[feb4a4f4d9146cdf]::traits::FulfillmentError>>::evaluate_obligations_error_on_ambiguity
  23:     0x7f2ad66be1b8 - rustc_hir_analysis[37999fc34f267976]::check::wfcheck::check_trait
  24:     0x7f2ad646dc8d - rustc_hir_analysis[37999fc34f267976]::check::check::check_item_type
  25:     0x7f2ad64663f6 - rustc_hir_analysis[37999fc34f267976]::check::wfcheck::check_well_formed
  26:     0x7f2ad64663d7 - rustc_query_impl[1b717c0605046cd0]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[1b717c0605046cd0]::query_impl::check_well_formed::dynamic_query::{closure#2}::{closure#0}, rustc_middle[829e5443847148d]::query::erase::Erased<[u8; 1usize]>>
  27:     0x7f2ad6465c4d - rustc_query_system[5c35911b3dd767e9]::query::plumbing::try_execute_query::<rustc_query_impl[1b717c0605046cd0]::DynamicConfig<rustc_data_structures[4e9b1f6e439fd3ef]::vec_cache::VecCache<rustc_span[14443e5baae9ab13]::def_id::LocalDefId, rustc_middle[829e5443847148d]::query::erase::Erased<[u8; 1usize]>, rustc_query_system[5c35911b3dd767e9]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[1b717c0605046cd0]::plumbing::QueryCtxt, false>
  28:     0x7f2ad6465756 - rustc_query_impl[1b717c0605046cd0]::query_impl::check_well_formed::get_query_non_incr::__rust_end_short_backtrace
  29:     0x7f2ad6462ef0 - rustc_hir_analysis[37999fc34f267976]::check::wfcheck::check_type_wf
  30:     0x7f2ad6462ddd - rustc_query_impl[1b717c0605046cd0]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[1b717c0605046cd0]::query_impl::check_type_wf::dynamic_query::{closure#2}::{closure#0}, rustc_middle[829e5443847148d]::query::erase::Erased<[u8; 1usize]>>
  31:     0x7f2ad6fe3fc8 - rustc_query_system[5c35911b3dd767e9]::query::plumbing::try_execute_query::<rustc_query_impl[1b717c0605046cd0]::DynamicConfig<rustc_query_system[5c35911b3dd767e9]::query::caches::SingleCache<rustc_middle[829e5443847148d]::query::erase::Erased<[u8; 1usize]>>, false, false, false>, rustc_query_impl[1b717c0605046cd0]::plumbing::QueryCtxt, false>
  32:     0x7f2ad6fe3d8e - rustc_query_impl[1b717c0605046cd0]::query_impl::check_type_wf::get_query_non_incr::__rust_end_short_backtrace
  33:     0x7f2ad68703f4 - rustc_hir_analysis[37999fc34f267976]::check_crate
  34:     0x7f2ad6d4a19b - rustc_interface[eaf9095e0de5b742]::passes::analysis
  35:     0x7f2ad6d49df5 - rustc_query_impl[1b717c0605046cd0]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[1b717c0605046cd0]::query_impl::analysis::dynamic_query::{closure#2}::{closure#0}, rustc_middle[829e5443847148d]::query::erase::Erased<[u8; 0usize]>>
  36:     0x7f2ad6fe57a6 - rustc_query_system[5c35911b3dd767e9]::query::plumbing::try_execute_query::<rustc_query_impl[1b717c0605046cd0]::DynamicConfig<rustc_query_system[5c35911b3dd767e9]::query::caches::SingleCache<rustc_middle[829e5443847148d]::query::erase::Erased<[u8; 0usize]>>, false, false, false>, rustc_query_impl[1b717c0605046cd0]::plumbing::QueryCtxt, false>
  37:     0x7f2ad6fe5376 - rustc_query_impl[1b717c0605046cd0]::query_impl::analysis::get_query_non_incr::__rust_end_short_backtrace
  38:     0x7f2ad7195cd1 - <rustc_interface[eaf9095e0de5b742]::passes::create_and_enter_global_ctxt<core[4141bff0bf9ca9dd]::option::Option<rustc_interface[eaf9095e0de5b742]::queries::Linker>, rustc_driver_impl[5af842e777d449e6]::run_compiler::{closure#0}::{closure#2}>::{closure#2} as core[4141bff0bf9ca9dd]::ops::function::FnOnce<(&rustc_session[ea1eec3ed26e42d5]::session::Session, rustc_middle[829e5443847148d]::ty::context::CurrentGcx, alloc[83e7320d2ee5a6b1]::sync::Arc<rustc_data_structures[4e9b1f6e439fd3ef]::jobserver::Proxy>, &std[eaa43e369792efb9]::sync::once_lock::OnceLock<rustc_middle[829e5443847148d]::ty::context::GlobalCtxt>, &rustc_data_structures[4e9b1f6e439fd3ef]::sync::worker_local::WorkerLocal<rustc_middle[829e5443847148d]::arena::Arena>, &rustc_data_structures[4e9b1f6e439fd3ef]::sync::worker_local::WorkerLocal<rustc_hir[67c6535075053fb5]::Arena>, rustc_driver_impl[5af842e777d449e6]::run_compiler::{closure#0}::{closure#2})>>::call_once::{shim:vtable#0}
  39:     0x7f2ad706e398 - rustc_interface[eaf9095e0de5b742]::interface::run_compiler::<(), rustc_driver_impl[5af842e777d449e6]::run_compiler::{closure#0}>::{closure#1}
  40:     0x7f2ad6ef7e8e - std[eaa43e369792efb9]::sys::backtrace::__rust_begin_short_backtrace::<rustc_interface[eaf9095e0de5b742]::util::run_in_thread_with_globals<rustc_interface[eaf9095e0de5b742]::util::run_in_thread_pool_with_globals<rustc_interface[eaf9095e0de5b742]::interface::run_compiler<(), rustc_driver_impl[5af842e777d449e6]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>
  41:     0x7f2ad6ef7c60 - <std[eaa43e369792efb9]::thread::lifecycle::spawn_unchecked<rustc_interface[eaf9095e0de5b742]::util::run_in_thread_with_globals<rustc_interface[eaf9095e0de5b742]::util::run_in_thread_pool_with_globals<rustc_interface[eaf9095e0de5b742]::interface::run_compiler<(), rustc_driver_impl[5af842e777d449e6]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1} as core[4141bff0bf9ca9dd]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  42:     0x7f2ad6ef6038 - <std[eaa43e369792efb9]::sys::thread::unix::Thread>::new::thread_start
  43:     0x7f2ad0894ac3 - start_thread
                               at ./nptl/pthread_create.c:442:8
  44:     0x7f2ad09268c0 - __GI___clone3
                               at ./misc/../sysdeps/unix/sysv/linux/x86_64/clone3.S:81:0
  45:                0x0 - <unknown>


rustc version: 1.94.0-nightly (e29fcf45e 2026-01-04)
platform: x86_64-unknown-linux-gnu

query stack during panic:
#0 [check_well_formed] checking that `Foo` is well-formed
#1 [check_type_wf] checking that types are well-formed
#2 [analysis] running analysis passes on crate `bug4`
end of query stack
thread 'rustc' panicked at /rustc-dev/e29fcf45e4ae686d77b490bf07320f0d3a2cf35f/compiler/rustc_trait_selection/src/traits/specialize/mod.rs:133:21:
Box<dyn Any>
stack backtrace:
   0:     0x7f2ad5819c0b - <std[eaa43e369792efb9]::backtrace::Backtrace>::create
   1:     0x7f2ad5819b55 - <std[eaa43e369792efb9]::backtrace::Backtrace>::force_capture
   2:     0x7f2ad48115b7 - std[eaa43e369792efb9]::panicking::update_hook::<alloc[83e7320d2ee5a6b1]::boxed::Box<rustc_driver_impl[5af842e777d449e6]::install_ice_hook::{closure#1}>>::{closure#0}
   3:     0x7f2ad582c4b2 - std[eaa43e369792efb9]::panicking::panic_with_hook
   4:     0x7f2ad484de81 - std[eaa43e369792efb9]::panicking::begin_panic::<rustc_errors[3cfa44cd230a06eb]::ExplicitBug>::{closure#0}
   5:     0x7f2ad483e4e6 - std[eaa43e369792efb9]::sys::backtrace::__rust_end_short_backtrace::<std[eaa43e369792efb9]::panicking::begin_panic<rustc_errors[3cfa44cd230a06eb]::ExplicitBug>::{closure#0}, !>
   6:     0x7f2ad483e11c - std[eaa43e369792efb9]::panicking::begin_panic::<rustc_errors[3cfa44cd230a06eb]::ExplicitBug>
   7:     0x7f2ad486bd21 - <rustc_errors[3cfa44cd230a06eb]::diagnostic::BugAbort as rustc_errors[3cfa44cd230a06eb]::diagnostic::EmissionGuarantee>::emit_producing_guarantee
   8:     0x7f2ad4e19ea9 - rustc_middle[829e5443847148d]::util::bug::opt_span_bug_fmt::<rustc_span[14443e5baae9ab13]::span_encoding::Span>::{closure#0}
   9:     0x7f2ad4e1a032 - rustc_middle[829e5443847148d]::ty::context::tls::with_opt::<rustc_middle[829e5443847148d]::util::bug::opt_span_bug_fmt<rustc_span[14443e5baae9ab13]::span_encoding::Span>::{closure#0}, !>::{closure#0}
  10:     0x7f2ad4e0b65b - rustc_middle[829e5443847148d]::ty::context::tls::with_context_opt::<rustc_middle[829e5443847148d]::ty::context::tls::with_opt<rustc_middle[829e5443847148d]::util::bug::opt_span_bug_fmt<rustc_span[14443e5baae9ab13]::span_encoding::Span>::{closure#0}, !>::{closure#0}, !>
  11:     0x7f2ad2863aa4 - rustc_middle[829e5443847148d]::util::bug::bug_fmt
  12:     0x7f2ad6097fcd - rustc_trait_selection[feb4a4f4d9146cdf]::traits::project::confirm_candidate
  13:     0x7f2ad6077c8b - rustc_trait_selection[feb4a4f4d9146cdf]::traits::project::normalize_projection_term
  14:     0x7f2ad60758de - <rustc_trait_selection[feb4a4f4d9146cdf]::traits::normalize::AssocTypeNormalizer as rustc_type_ir[241df8ba430ad1e5]::fold::TypeFolder<rustc_middle[829e5443847148d]::ty::context::TyCtxt>>::fold_ty
  15:     0x7f2ad66a0f03 - rustc_ty_utils[6308f146fe78f831]::ty::impl_self_is_guaranteed_unsized
  16:     0x7f2ad66a07e1 - rustc_query_impl[1b717c0605046cd0]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[1b717c0605046cd0]::query_impl::impl_self_is_guaranteed_unsized::dynamic_query::{closure#2}::{closure#0}, rustc_middle[829e5443847148d]::query::erase::Erased<[u8; 1usize]>>
  17:     0x7f2ad645a7a3 - rustc_query_system[5c35911b3dd767e9]::query::plumbing::try_execute_query::<rustc_query_impl[1b717c0605046cd0]::DynamicConfig<rustc_query_system[5c35911b3dd767e9]::query::caches::DefIdCache<rustc_middle[829e5443847148d]::query::erase::Erased<[u8; 1usize]>>, false, false, false>, rustc_query_impl[1b717c0605046cd0]::plumbing::QueryCtxt, false>
  18:     0x7f2ad645a052 - rustc_query_impl[1b717c0605046cd0]::query_impl::impl_self_is_guaranteed_unsized::get_query_non_incr::__rust_end_short_backtrace
  19:     0x7f2ad646872f - rustc_hir_analysis[37999fc34f267976]::check::check::check_item_type
  20:     0x7f2ad64663f6 - rustc_hir_analysis[37999fc34f267976]::check::wfcheck::check_well_formed
  21:     0x7f2ad64663d7 - rustc_query_impl[1b717c0605046cd0]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[1b717c0605046cd0]::query_impl::check_well_formed::dynamic_query::{closure#2}::{closure#0}, rustc_middle[829e5443847148d]::query::erase::Erased<[u8; 1usize]>>
  22:     0x7f2ad6465c4d - rustc_query_system[5c35911b3dd767e9]::query::plumbing::try_execute_query::<rustc_query_impl[1b717c0605046cd0]::DynamicConfig<rustc_data_structures[4e9b1f6e439fd3ef]::vec_cache::VecCache<rustc_span[14443e5baae9ab13]::def_id::LocalDefId, rustc_middle[829e5443847148d]::query::erase::Erased<[u8; 1usize]>, rustc_query_system[5c35911b3dd767e9]::dep_graph::graph::DepNodeIndex>, false, false, false>, rustc_query_impl[1b717c0605046cd0]::plumbing::QueryCtxt, false>
  23:     0x7f2ad6465756 - rustc_query_impl[1b717c0605046cd0]::query_impl::check_well_formed::get_query_non_incr::__rust_end_short_backtrace
  24:     0x7f2ad6462ef0 - rustc_hir_analysis[37999fc34f267976]::check::wfcheck::check_type_wf
  25:     0x7f2ad6462ddd - rustc_query_impl[1b717c0605046cd0]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[1b717c0605046cd0]::query_impl::check_type_wf::dynamic_query::{closure#2}::{closure#0}, rustc_middle[829e5443847148d]::query::erase::Erased<[u8; 1usize]>>
  26:     0x7f2ad6fe3fc8 - rustc_query_system[5c35911b3dd767e9]::query::plumbing::try_execute_query::<rustc_query_impl[1b717c0605046cd0]::DynamicConfig<rustc_query_system[5c35911b3dd767e9]::query::caches::SingleCache<rustc_middle[829e5443847148d]::query::erase::Erased<[u8; 1usize]>>, false, false, false>, rustc_query_impl[1b717c0605046cd0]::plumbing::QueryCtxt, false>
  27:     0x7f2ad6fe3d8e - rustc_query_impl[1b717c0605046cd0]::query_impl::check_type_wf::get_query_non_incr::__rust_end_short_backtrace
  28:     0x7f2ad68703f4 - rustc_hir_analysis[37999fc34f267976]::check_crate
  29:     0x7f2ad6d4a19b - rustc_interface[eaf9095e0de5b742]::passes::analysis
  30:     0x7f2ad6d49df5 - rustc_query_impl[1b717c0605046cd0]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[1b717c0605046cd0]::query_impl::analysis::dynamic_query::{closure#2}::{closure#0}, rustc_middle[829e5443847148d]::query::erase::Erased<[u8; 0usize]>>
  31:     0x7f2ad6fe57a6 - rustc_query_system[5c35911b3dd767e9]::query::plumbing::try_execute_query::<rustc_query_impl[1b717c0605046cd0]::DynamicConfig<rustc_query_system[5c35911b3dd767e9]::query::caches::SingleCache<rustc_middle[829e5443847148d]::query::erase::Erased<[u8; 0usize]>>, false, false, false>, rustc_query_impl[1b717c0605046cd0]::plumbing::QueryCtxt, false>
  32:     0x7f2ad6fe5376 - rustc_query_impl[1b717c0605046cd0]::query_impl::analysis::get_query_non_incr::__rust_end_short_backtrace
  33:     0x7f2ad7195cd1 - <rustc_interface[eaf9095e0de5b742]::passes::create_and_enter_global_ctxt<core[4141bff0bf9ca9dd]::option::Option<rustc_interface[eaf9095e0de5b742]::queries::Linker>, rustc_driver_impl[5af842e777d449e6]::run_compiler::{closure#0}::{closure#2}>::{closure#2} as core[4141bff0bf9ca9dd]::ops::function::FnOnce<(&rustc_session[ea1eec3ed26e42d5]::session::Session, rustc_middle[829e5443847148d]::ty::context::CurrentGcx, alloc[83e7320d2ee5a6b1]::sync::Arc<rustc_data_structures[4e9b1f6e439fd3ef]::jobserver::Proxy>, &std[eaa43e369792efb9]::sync::once_lock::OnceLock<rustc_middle[829e5443847148d]::ty::context::GlobalCtxt>, &rustc_data_structures[4e9b1f6e439fd3ef]::sync::worker_local::WorkerLocal<rustc_middle[829e5443847148d]::arena::Arena>, &rustc_data_structures[4e9b1f6e439fd3ef]::sync::worker_local::WorkerLocal<rustc_hir[67c6535075053fb5]::Arena>, rustc_driver_impl[5af842e777d449e6]::run_compiler::{closure#0}::{closure#2})>>::call_once::{shim:vtable#0}
  34:     0x7f2ad706e398 - rustc_interface[eaf9095e0de5b742]::interface::run_compiler::<(), rustc_driver_impl[5af842e777d449e6]::run_compiler::{closure#0}>::{closure#1}
  35:     0x7f2ad6ef7e8e - std[eaa43e369792efb9]::sys::backtrace::__rust_begin_short_backtrace::<rustc_interface[eaf9095e0de5b742]::util::run_in_thread_with_globals<rustc_interface[eaf9095e0de5b742]::util::run_in_thread_pool_with_globals<rustc_interface[eaf9095e0de5b742]::interface::run_compiler<(), rustc_driver_impl[5af842e777d449e6]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>
  36:     0x7f2ad6ef7c60 - <std[eaa43e369792efb9]::thread::lifecycle::spawn_unchecked<rustc_interface[eaf9095e0de5b742]::util::run_in_thread_with_globals<rustc_interface[eaf9095e0de5b742]::util::run_in_thread_pool_with_globals<rustc_interface[eaf9095e0de5b742]::interface::run_compiler<(), rustc_driver_impl[5af842e777d449e6]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1} as core[4141bff0bf9ca9dd]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  37:     0x7f2ad6ef6038 - <std[eaa43e369792efb9]::sys::thread::unix::Thread>::new::thread_start
  38:     0x7f2ad0894ac3 - start_thread
                               at ./nptl/pthread_create.c:442:8
  39:     0x7f2ad09268c0 - __GI___clone3
                               at ./misc/../sysdeps/unix/sysv/linux/x86_64/clone3.S:81:0
  40:                0x0 - <unknown>


query stack during panic:
#0 [impl_self_is_guaranteed_unsized] computing whether `<impl at ../../try_pro/bug4.rs:13:1: 13:35>` has a guaranteed unsized self type
#1 [check_well_formed] checking that `<impl at ../../try_pro/bug4.rs:13:1: 13:35>` is well-formed
#2 [check_type_wf] checking that types are well-formed
#3 [analysis] running analysis passes on crate `bug4`
end of query stack

```

</p>
</details>
